From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Sun, 16 Aug 2026 14:19:45 +0200
Subject: fix(devicetree-firmware): include soc specific firmwares in
 install_generic()

Qualcomm boards may use SoC specific firmwares from an otherwise generic
driver like qcom-geni-se. Which firmware to load is specified with
a firmware-name devicetree-property, so install_hostonly() already picks
these up.

Make install_generic() also include these in the initramfs by explicitly
searching for them similar to how this is already done for Qualcomm laptop
model specific firmwares.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/48e66eada503033c4d9e6fcd6f195a40f5610c30
---
 modules.d/70devicetree-firmware/module-setup.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules.d/70devicetree-firmware/module-setup.sh b/modules.d/70devicetree-firmware/module-setup.sh
index 16b9e00..e725ac9 100755
--- a/modules.d/70devicetree-firmware/module-setup.sh
+++ b/modules.d/70devicetree-firmware/module-setup.sh
@@ -57,6 +57,12 @@ install_generic() {
                 _fw_files+=("$_fw")
             done
         done
+        # add soc specific firmwares needed to boot some qcom boards
+        for _fw in "$_fwdir/qcom"/*/qupv3fw.elf*; do
+            [ -f "$_fw" ] || continue
+            _fw=${_fw#"${dracutsysrootdir-}"}
+            _fw_files+=("$_fw")
+        done
     done
     ((${#_fw_files[@]} > 0)) && inst_multiple -o "${_fw_files[@]}"
 }
